drivers/usb/host/xhci-ring.c

```diff
@@ -541,7 +544,27 @@ void xhci_find_new_dequeue_state(struct xhci_hcd *xhci,
 	hw_dequeue = xhci_get_hw_deq(xhci, dev, ep_index, stream_id);
 	new_seg = ep_ring->deq_seg;
 	new_deq = ep_ring->dequeue;
-	state->new_cycle_state = hw_dequeue & 0x1;
+
+	if (!(ep->ep_state & EP_HAS_STREAMS))
+		halted_seg = trb_in_td(xhci, cur_td->start_seg,
```

Comment:
It seems wasteful to calculate this regardless of whether or not the quirk is set.

Reply:
I wanted to avoid a somewhat awkward control flow -  
```
if(quirk && !ep_has_streams) {
    halted_seg = trb_in_td();
    if (halted_seg)
        use_trb_dcs;
    else
        use_hw_dcs;
} else {
    use_hw_dcs;
}
```
but if this has a lighter touch when the quirk is not set then I suppose it's better.

Reply:
I'd be happy with:
```
if(quirk && !ep_has_streams)
    halted_seg = trb_in_td();
if (halted_seg)
    use_trb_dcs;
else
    use_hw_dcs;
```